### ISSUE TITLE
Cannot set property id of undefined when importing a collection

### ISSUE THREAD

**Issue by alex88** [a85/POSTMan-Chrome-Extension#8538]
**Describe the bug**
Importing a collection just exported from postman doesn't work

**To Reproduce**
1. Exportt a collection
2. Import it again
3. See error

**Expected behavior**
I doesn't throw an error

**App information (please complete the following information):**
 - Native App
 - 7.25.0
 - macOS Catalina 10.15.4

**Additional context**
Logs show:

```
[10054][1590403471470][requester][error]["Error while importing the collection",{"name":"TypeError","message":"Cannot set property 'id' of undefined","stack":"TypeError: Cannot set property 'id' of undefined\n    at nodeEvents.forEach (file:///Applications/Postman.app/Contents/Resources/app/js/vendor-shared.js:211719:15)\n    at Array.forEach (<anonymous>)\n    at __cloneNode (file:///Applications/Postman.app/Contents/Resources/app/js/vendor-shared.js:211718:28)\n    at __cloneTree (file:///Applications/Postman.app/Contents/Resources/app/js/vendor-shared.js:211873:10)\n    at Promise.all.children.map.child (file:///Applications/Postman.app/Contents/Resources/app/js/vendor-shared.js:211878:14)\n    at Array.map (<anonymous>)\n    at __cloneNode.then (file:///Applications/Postman.app/Contents/Resources/app/js/vendor-shared.js:211876:33)"}]
```

It seems that I need to be connected to the right workspace for it to work, however I don't find a way to "unlink" the json from the specific workspace, is there a workaround?

**Comment by cyrielle-hada** [comment]
@alex88 I wasn't able to reproduce the behavior on Postman v.7.25. Once the collection exported, you should be able to import it to any workspace.

Would it be possible [to raise a support ticket](https://go.pstmn.io/help-center-signin-newrequest) and send your logs as well as a screen record that shows how to replicate the behavior?

**Comment by alex88** [comment]
@cyrielle-hada thanks for replying! I've opened support ticket 51374

**Comment by xk0der** [comment]
Hey @alex88, closing the GitHub issue too as the root-cause was found and workaround shared at the internal support ticket.

**Comment by alex88** [comment]
Thanks @xk0der!
Just as a reference, if anyone else encounter the same issue, double check you json "events" field in each collection folder. Try to set test/pre-request scripts to something that's not an empty string, update, then back to an empty script.

**Comment by xk0der** [comment]
Thanks for that, @alex88 !

**To add to @alex88 message.**
The workaround is very specific to the root-cause we found. Here a "script" node was missing the remaining nodes which would have contained the **Pre-request** and **test** scripts.

Unfortunately, the error message is a generic one and maybe thrown for various other reasons too. For most Postman users they should not run into this issue, as the exported JSON should be generated fine. 

In this particular use-case, the exported JSON was being modified manually and imported back. 

Our first hunch is that an automated git-merge caused the JSON to be malformed, and a potentially less strict importer in Postman imported this Collection JSON in earlier versions. Now that the importer has been improved, it is catching these malformed entries in the Collection JSON. 

Another hypothesis is that there may have been a bug in Postman exporter in an older version, and the subsequent export and import messed up the Collection structure. Which is now being caught after Exporter and Import both have been considerably improved.

**TL;DR**
* If you are using Postman to manage your Collections and not editing the exported JSON outside of Postman, you should not run into this issue.
* If you do encounter a similar, check the JSON structure for missing elements. And using that as a cue, fix the required requests/folders within the Collection.
* If nothing works, reach out to Postman support: https://www.postman.com/support/
